Dockerfile

```diff
@@ -241,7 +241,13 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
 COPY ./contrib/dockerd-rootless.sh /build
 COPY ./contrib/dockerd-rootless-setuptool.sh /build
 
-FROM djs55/vpnkit:${VPNKIT_VERSION} AS vpnkit
+FROM --platform=amd64 djs55/vpnkit:${VPNKIT_VERSION} AS vpnkit-amd64
+
+FROM --platform=arm64 djs55/vpnkit:${VPNKIT_VERSION} AS vpnkit-arm64
+
+FROM scratch AS vpnkit
+COPY --from=vpnkit-amd64 /vpnkit /build/vpnkit.x86_64
+COPY --from=vpnkit-arm64 /vpnkit /build/vpnkit.aarch64
```

Comment:
Why are we putting multiple binaries into a stage, including binary for wrong arch? Just the one for the target platform should suffice.

Reply:
Ideally we could like `COPY --from=vpnkit-${TARGETPLATFORM} /vpnkit /build/vpnkit`, but we can't, because `djs55/vpnkit:0.5.0` only contains `amd64` and `arm64` manifests.